demo/dotnet/LeopardDemo/MicDemo.cs

```diff
@@ -0,0 +1,206 @@
+﻿/*
+    Copyright 2020-2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Pv;
+
+namespace LeopardDemo
+{
+    /// <summary>
+    /// Microphone Demo for Leopard Speech-to-Tntent engine. It creates an input audio stream from a microphone. 
+    /// </summary>
+    public class MicDemo
+    {
+
+        private static readonly int PV_RECORDER_FRAME_LENGTH = 2048;
+
+        /// <summary>
+        /// Creates an input audio stream and instantiates an instance of Leopard object.
+        /// </summary>
+        /// <param name="accessKey">AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).</param>
+        /// <param name="modelPath">Absolute path to the file containing model parameters. If not set it will be set to the default location.</param>           
+        /// <param name="audioDeviceIndex">Optional argument. If provided, audio is recorded from this input device. Otherwise, the default audio input device is used.</param>        
+        public static void RunDemo(
+            string accessKey,
+            string modelPath,
+            int audioDeviceIndex)
+        {
+
+            Leopard Leopard = null;
+
+            Leopard = Leopard.Create(
+                accessKey,
+                modelPath);
+
+            PvRecorder recorder = PvRecorder.Create(audioDeviceIndex, PV_RECORDER_FRAME_LENGTH);
+
+            List<short> audioFrame = new List<short>();
+
+            var tokenSource = new CancellationTokenSource();
+            CancellationToken token = tokenSource.Token;
+
+            Console.CancelKeyPress += (s, o) =>
+            {
+                Console.WriteLine("Stopping...");
+
+                tokenSource.Cancel();
+                Leopard?.Dispose();
+                recorder.Dispose();
+            };
+
+            Task recordingTask = Task.Run(() => {
+
+                token.ThrowIfCancellationRequested();
+
+                audioFrame.Clear();
+                recorder.Start();
+                while(!token.IsCancellationRequested)
+                {
+                    short[] pcm = recorder.Read();
+                    audioFrame.AddRange(pcm);
+                }
+                recorder.Stop();
+            }, tokenSource.Token);
```

Comment:
I don't think you need to create a new token
```suggestion
            }, token);
```